src/control/Tool.cpp

```diff
@@ -12,6 +12,24 @@ Tool::Tool(string name, ToolType type, Color color, int capabilities, double* th
     setColor(color);
 }
 
+Tool::Tool(Tool* t) {
+    this->name = t->name;
+    this->type = t->type;
+    if (t->thickness) {
+        this->thickness = new double[5];
```

Comment:
Nice! Thanks for sharing that link.
This actually also helps to get that magic number out of the `ToolHandler.cpp`.

One question though:  Right now I've added this as a `public static int toolSizes = 5` to `Tool`, which in this case makes sense. If it would be a generic const. what the bes practice to namespace it to avoid naming conflicts?

```cpp
class Tool: public ToolBase {
public:
    Tool(string name, ToolType type, Color color, int capabilities, double* thickness);
    Tool(Tool* t);
    virtual ~Tool();

    static const int toolSizes = 5;
```

Reply:
`static const int` makes here most sense as the variable is actually strongly connected to the class. Hence it should be a member of the class. You would make it a `const int` without the class association if it would be some global array. In the end it will be handled the same way by the compiler (shifted to the text section of the binary). So your solution is best in our case.